src/main/java/org/sagebionetworks/web/client/widget/header/Header.java

```diff
@@ -50,32 +57,45 @@ public Widget asWidget() {
 		view.setPresenter(this);
 		return view.asWidget();
 	}
-	
+
 	public void setSearchVisible(boolean searchVisible) {
 		view.setSearchVisible(searchVisible);
 	}
-	
+
 	public void refresh() {
+		UserSessionData userSessionData = authenticationController.getCurrentUserSessionData();
+		if (cachedUserSessionData == null || cachedUserSessionData.equals(userSessionData)) {
```

Comment:
Should only get into if null or user change.  Should contain init favorites call (not get favorites on every refresh).